app/src/main/java/com/hover/stax/data/repository/AccountRepositoryImpl.kt

```diff
@@ -28,22 +30,25 @@ class AccountRepositoryImpl(val accountRepo: AccountRepo, val channelRepo: Chann
     override val fetchAccounts: Flow<List<Account>>
         get() = accountRepo.getAccounts()
 
-    override suspend fun createAccounts(channels: List<Channel>): List<Long> {
+    override suspend fun createAccounts(channels: List<Channel>) {
         val defaultAccount = accountRepo.getDefaultAccountAsync()
+        channels.mapIndexed { index, channel ->
+            createAccount(channel, -1, defaultAccount == null && index == 0)}
+    }
 
-        val accounts = channels.mapIndexed { index, channel ->
-            val accountName: String = if (getFetchAccountAction(channel.id) == null) channel.name else channel.name.plus(PLACEHOLDER )//placeholder alias for easier identification later
-            Account(
-                accountName, channel.name, channel.logoUrl, channel.accountNo, channel.id, channel.countryAlpha2,
-                channel.id, channel.primaryColorHex, channel.secondaryColorHex, defaultAccount == null && index == 0
-            )
-        }.onEach {
-            logChoice(it)
-            ActionApi.scheduleActionConfigUpdate(it.countryAlpha2, 24, context)
-        }
+    override suspend fun createAccount(channel: Channel, subscriptionId: Int, isDefault: Boolean) {
```

Comment:
Things to note on this function:
- The account name allocation needs to be changed to match the one in `ChannelsViewModel`.
- Generally, batch inserts are faster than multiple single inserts. 
- Batch inserts are atomic, meaning if one insert/update fails, the entire transaction is rolled back, depending on use case, this may be a desired effect.